src/main/java/com/minecolonies/coremod/commands/generalcommands/ScanCommand.java

```diff
@@ -52,41 +51,47 @@ public ScanCommand(@NotNull final String... parents)
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu) throws CommandException
     {
         // Will throw ClassCastException if null, but should never be null as these values are required.
+        final EntityPlayerMP player = actionMenu.getPlayerForArgument("player");
         final int x1 = actionMenu.getIntegerForArgument("x1");
         final int y1 = actionMenu.getIntegerForArgument("y1");
         final int z1 = actionMenu.getIntegerForArgument("z1");
         final int x2 = actionMenu.getIntegerForArgument("x2");
         final int y2 = actionMenu.getIntegerForArgument("y2");
         final int z2 = actionMenu.getIntegerForArgument("z2");
+
         final BlockPos from = new BlockPos(x1, y1, z1);
         final BlockPos to = new BlockPos(x2, y2, z2);
-        executeShared(server, sender, from, to);
+        executeShared(server, sender, from, to, player);
     }
 
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
     {
-//        executeShared(server, sender, from, to);
+        //executeShared(server, sender, from, to);
```

Comment:
If you are not going to support the old-style command, delete the method and remove "extends AbstractSingleCommand".
You could also merge executeShared into execute.

Reply:
If I remove abstractSingleCammd is removes a bunch of other methods as well, as well as "isOpped" is that correct?

Reply:
Yes, that might be the case, but I think we'd be better off using a new file that's specific to the new command structure.   Also, there's no reason to use inheritance to provide those methods vs using a utility class.

For "isOpped", this might be your chance to add a forge permission instead, since we have the framework in place to support that.